src/Agent.Worker/TestResults/ResultsCommandExtension.cs

```diff
@@ -236,7 +238,7 @@ private void ProcessPublishTestResultsCommand(IExecutionContext context, Diction
                     {
                         cancellationToken.ThrowIfCancellationRequested();
                         string runName = null;
-                        if (!string.IsNullOrWhiteSpace(_runTitle))
+                        if (!string.IsNullOrWhiteSpace(_runTitle) && resultFiles.Count > 1)
```

Comment:
why this is needed

Reply:
We were adding _{count} to the custom test run title when there's more than on file. So I am just avoiding it adding when there's only one file/run. Currently it's the case for VsTest.